cmd/textiled/main.go

```diff
@@ -212,7 +241,14 @@ var rootCmd = &cobra.Command{
 		addrGatewayHost := cmd.AddrFromStr(configViper.GetString("addr.gateway.host"))
 		addrGatewayUrl := configViper.GetString("addr.gateway.url")
 
-		addrFilecoinApi := cmd.AddrFromStr(configViper.GetString("addr.filecoin.api"))
+		var addrFilecoinApi ma.Multiaddr
+		if filStr := configViper.GetString("addr.filecoin.api"); filStr != "" {
+			addrFilecoinApi = cmd.AddrFromStr(filStr)
+		}
+
```

Comment:
@asutula i needed to add this step so you could pass an empty address and not run api (as there is a nil check later that could never have been nil afai can tell)

Reply:
Nice. I should have done it like this anyway. Good call.